test/fges.jl

```diff
@@ -0,0 +1,44 @@
+using Distributions, CausalInference, Graphs
+using Test, Random
+
+
+Random.seed!(100)
+@testset "GES " begin
+    for n in 0:10
+        alpha = 0.1
+        @testset "randdag($n)" begin for k in 1:100
+            global g = randdag(n, alpha)
+            h2 = cpdag(g)
+            h1 = CausalInference.fges_internal(g, Float64, h2)
+        
+            h1 == h2 || println(vpairs(g))
+            @test vpairs(h1) ⊆ vpairs(h2)
```

Comment:
This holds so it looks like there are indeed new colliders

Reply:
I think it might be true that the score function for a vertex and its parents has to depend on the whole graph in an oracle setting (not only on the edges between v and its parents). 

In Probabilistic Graphical Models by Koller/Friedman (one can find the pdf of this book online rather easily) Cor. 18.1 on page 793 seems to suggest to me that something like counting the number of vertices p coming before v in a topological ordering for which p is not independent of v given its parents (this can be tested by a d-sep oracle in the true DAG) might be possible. The underlying idea here goes into the direction of counting the violations of the Markov property.

E.g. for CPDAG a - b - c, we have for the wrong DAG a -> b <- c and top. ordering a, c, b that c is not independent from vertex a given the empty set in the "true" DAGs represented by the CPDAG.

Whereas for a "correct" DAG a -> b -> c, we e.g. have that c indep a given b. Here no Markov property is violated.

This is also the case for every complete DAG, e.g. a -> b -> c with a -> c.
So, we would also need to have a penalty term regarding the model complexity for this score. 

There might be something easier, I'm not an expert in score based approaches.